test/functional/test-amp-context.js

```diff
@@ -0,0 +1,316 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  AmpContext,
+} from '../../3p/ampcontext';
+import {MessageType} from '../../src/3p-frame-messaging';
+import * as sinon from 'sinon';
+
+describe('3p ampcontext.js', () => {
+  let windowPostMessageSpy;
+  let windowMessageHandler;
+  let win;
+  let sandbox;
+
+  beforeEach(() => {
+    sandbox = sinon.sandbox.create();
+    windowPostMessageSpy = sinon.sandbox.spy();
+    win = {
+      addEventListener: (eventType, handlerFn) => {
+        expect(eventType).to.equal('message');
+        expect(windowMessageHandler).to.not.be.ok;
+        windowMessageHandler = handlerFn;
+      },
+      parent: {
+        postMessage: windowPostMessageSpy,
+      },
+    };
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+    win = undefined;
+    windowMessageHandler = undefined;
+  });
+
+  it('should add metadata to window.context using name.', () => {
+    win.name = generateStringifiedAttributes();
```

Comment:
o.O
Good catch. That is actually a bigger problem than just for this PR, the AmpContext library is parsing improperly overall. I have no idea how I missed this. Will submit a separate PR to fix this ASAP (and will also address the local issues in this PR).